crates/sc-proof-of-time/src/node_client.rs

```diff
@@ -79,14 +65,43 @@ where
         error!("pot_client: gossip engine has terminated.");
     }
 
+    /// Initializes the chain state from the consensus tip info.
+    async fn initialize(&self) {
+        // Wait for a block with proofs.
+        info!("pot_client::initialize: waiting for initialization ...");
+        let delay = tokio::time::Duration::from_secs(1);
+        let proofs = loop {
+            let tip = get_consensus_tip(
+                self.client.clone(),
+                self.sync_oracle.clone(),
+                self.chain_info_fn.clone(),
+            )
+            .await
+            .expect("Consensus tip info should be available");
```

Comment:
This is not a proof that it will be though. I expect proofs in `.expect()`. The same not proof is also used in other places.

Reply:
Expectations are not proofs. `.expect()` must contain a proof why it will never ever panic. I have explained this many times at this point.

Reply:
If it crashes here because of any of the 3 reasons that should not happen (unable to read hdr from  backend, etc), this is what I would want to see: reason why it crashed, other things are cosmetic. I don't understand your notion of "proof" at this point (I also checked the other instances, this looks consistent). Dragging this further is not productive, I can do this as a follow up if needed.

Reply:
It shouldn't happen, but it doesn't mean it will not happen and there is nothing this function can prove it will not happen. If you can't prove it - `.expect()` is not appropriate for this situation. I'm just concerned with increased number of `.expect()` that are not proofs in our codebase. I think I'll have to go through all of them in proof of time crates and either fix or add TODOs to fix.

If there could be error, `fn initialize` should return `Result<T, E>`. There are a few invalid proofs in `sc-consensus-subspace` I'm aware of that I have added too, some might even have TODOs, but we should be very careful with these in the code.

Please add TODOs to handle these properly in the future.

Reply:
Yeah that makes sense. I tend to add `expect()` only when a precondition fails and we cannot proceed further (e.g) failed to read from client backend, failed to create thread, etc or it is something we would address in an immediate future PR. 

I reviewed the `expect()` in the crate, think there is only one instance (the config could be bad and PoT instance could not be created) that can be handled, for rest it is better to halt IMO. But added TODOs so we don't lose track